src/game.cpp

```diff
@@ -11153,22 +11153,15 @@ void game::butcher()
         u.assign_activity( ACT_LONGSALVAGE, 0, salvage_tool_index );
         return;
     }
-    const item &dis_item = items[corpses[butcher_corpse_index]];
-    if( !dis_item.is_corpse() && butcher_corpse_index < (int)salvage_index) {
-        const recipe *cur_recipe = get_disassemble_recipe(dis_item.type->id);
-        assert(cur_recipe != NULL); // tested above
-        if( !query_dissamble( dis_item ) ) {
-            return;
-        }
-        u.assign_activity(ACT_DISASSEMBLE, cur_recipe->time, cur_recipe->id);
-        u.activity.values.push_back(corpses[butcher_corpse_index]);
-        u.activity.values.push_back(1);
+    item *dis_item = &items[corpses[butcher_corpse_index]];
```

Comment:
Why a pointer, can't you keep it as reference? That would require less code changes below (-;

Consequently, the new `disassemble` should take the item as reference, it makes no sense to supply a `nullptr` for it, so there is no advantages of using a pointer.